scripts/get-tokens-decimals-not-8.ts

```diff
@@ -0,0 +1,139 @@
+
+const getAgent = (): Promise<HttpAgent> =>
+  createAgent({
+    identity: new AnonymousIdentity(),
+    host: "https://icp-api.io",
+  });
+
+const orchestratorInfo = async ({
+  orchestratorId: canisterId,
+}: {
+  orchestratorId: Principal;
+}): Promise<OrchestratorInfo> => {
+  const agent = await getAgent();
+  const { getOrchestratorInfo } = CkETHOrchestratorCanister.create({
+    agent,
+    canisterId,
+  });
+
+  return getOrchestratorInfo({ certified: true });
+};
+
+type TokenInfo = {
+  ledgerCanisterId: string;
+  tokenSymbol: string;
+  decimals?: bigint;
+};
+
+const buildOrchestratorInfo = async (
+  orchestratorId: Principal
+): Promise<TokenInfo[]> => {
+  const { managed_canisters } = await orchestratorInfo({ orchestratorId });
+
+  // eslint-disable-next-line local-rules/prefer-object-params -- This is a destructuring assignment
+  const mapManagedCanisters = ({
+    ledger,
+    ckerc20_token_symbol,
+  }: ManagedCanisters): TokenInfo | undefined => {
+    const ledgerCanister = fromNullable(ledger);
+
+    // Skip tokens without Ledger or Index (by definition, this can happen).
+    if (isNullish(ledgerCanister)) {
+      return undefined;
+    }
+
+    const { canister_id: ledgerCanisterId } =
+      "Created" in ledgerCanister
+        ? ledgerCanister.Created
+        : ledgerCanister.Installed;
+
+    return {
+      ledgerCanisterId: ledgerCanisterId.toText(),
+      tokenSymbol: ckerc20_token_symbol,
+    };
+  };
+
+  const assertToken = (token: TokenInfo | undefined): token is TokenInfo =>
+    token !== undefined;
+
+  return managed_canisters.map(mapManagedCanisters).filter(assertToken);
+};
+
+const addTokenDecimals = async (token: TokenInfo): Promise<TokenInfo> => {
+  const { metadata } = IcrcLedgerCanister.create({
+    agent: await getAgent(),
+    canisterId: Principal.fromText(token.ledgerCanisterId),
+  });
+
+  const data = await metadata({ certified: true });
+
+  const decimalsEntry = data.find((item) => item[0] === "icrc1:decimals");
+  const decimals = "Nat" in decimalsEntry![1] ? decimalsEntry![1].Nat : 0n;
+
+  return {
+    ...token,
+    decimals,
+  };
+};
+
+const ORCHESTRATOR_STAGING_ID = Principal.fromText(
+  "2s5qh-7aaaa-aaaar-qadya-cai"
+);
+const ORCHESTRATOR_PRODUCTION_ID = Principal.fromText(
+  "vxkom-oyaaa-aaaar-qafda-cai"
+);
+
+const findCkErc20 = async () => {
+  const [staging, production] = await Promise.all(
+    [ORCHESTRATOR_STAGING_ID, ORCHESTRATOR_PRODUCTION_ID].map(
+      buildOrchestratorInfo
+    )
+  );
+
+  const ckETH: TokenInfo = {
+    ledgerCanisterId: "ss2fx-dyaaa-aaaar-qacoq-cai",
+    tokenSymbol: "ckETH",
+  };
+
+  const tokens = [ckETH, ...production, ...staging];
+
+  const tokensWithDecimals = await Promise.all(tokens.map(addTokenDecimals));
+  const tokensNot8Decimals = tokensWithDecimals.filter(
+    (token) => token.decimals !== 8n
+  );
+
+  writeFileSync(
+    "tokens.json",
+    JSON.stringify(tokensNot8Decimals, (_key, value) => {
```

Comment:
Not sure if you replace on purpose because you want that output format or just because the `stringify` complained. If the later, and just in case, you can use `jsonReplacer` from the utils.